packages/x-components/src/x-modules/facets-next/service/facets.service.ts

```diff
@@ -159,4 +163,40 @@ export class DefaultFacetsService implements FacetsService {
   protected setFacetGroup(facetGroup: FacetGroupEntry): void {
     this.store.commit('x/facetsNext/setFacetGroup', facetGroup);
   }
+
+  /**
+   * Removes the facets that belong to the given group.
+   *
+   * @param groupId - The id of the group from whom remove the facets that are in the store.
+   * @returns The removed facets.
+   * @internal
+   */
+  protected removeGroupFacets(groupId: FacetsGroup['id']): Omit<Facet, 'filters'>[] {
+    const facetsToRemove = Object.values(this.store.state.x.facetsNext.facets).filter(
+      facet => this.store.state.x.facetsNext.groups[facet.id] === groupId
+    );
+    facetsToRemove.forEach(this.removeFacet.bind(this));
+    return facetsToRemove;
```

Comment:
Just one question, whats the meaning of returning the facets that we are going to remove? I just wanna know it

Reply:
I think because it is a good practice to return the objects you actually remove, just in case you want to restore them, and for consistency with the `removeFilters` method.